angel_system/eval/visualization.py

```diff
@@ -19,17 +19,20 @@ def plot_activity_confidence(label, gt_ranges, det_ranges, output_dir, custom_ra
     :param custom_range_color: The color of the additional range to be drawn. If not set, we will
                                use "red".
     """
-    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(14, 6))
-    fig.suptitle(f"Window Confidence over time for \"{label}\"")
-    ax1.set_title("Real Detected Actions")
-    ax2.set_title("Optimal Detected Actions")
-    for ax in [ax1, ax2]:
-        ax.set_xlabel("Time (seconds)")
-        ax.set_ylabel("Confidence")
-        ax.set_ylim(0, 1.05)
-        ax.set_xlim(GlobalValues.slice_time_ranges[0][0] - 1,
-                    GlobalValues.slice_time_ranges[-1][1] + 1)
-        # plt.yscale("log")
+    all_start_times = [p["time"][0] for p in gt_ranges]
+    all_start_times.extend([p["time"][0] for p in det_ranges[label]])
+    all_end_times = [p["time"][1] for p in gt_ranges]
+    all_end_times.extend([p["time"][1] for p in det_ranges[label]])
+    min_start_time = min(all_start_times) - 1
+    max_end_time = max(all_end_times) + 1
```

Comment:
The +-1 part was somewhat hard-coded values from the example notebooks. Using something that is percentile based would be more stable for different length scales of the input data (e.g. pad 5% total time delta to the beginning and end of the plot).